### PR TITLE
SPR-40 Create API Routes

### PR DESCRIPTION
**Pull request by dbarkowsky**
# Description

This PR includes the following proposed change(s):

- Added GET endpoint to return all documents
- Added GET endpoint to get documents with specific IDIR
- Added GET endpoint to get single document with specific ID
- Added PATCH endpoint to change state property of document
- Offered query string for GET requests to only return a subset of properties. 
- Removed hardcoded http:// before hostnames
- Added bypass to rate limiter for testing

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation
- [ ] Version change
- [x] Dependencies added/removed

## Aspect(s) of Project Affected
- [ ] Frontend
- [x] API
- [ ] Database
- [ ] Workflows
- [x] Documentation

## Best Practices Checklist
- [x] I have performed a self-review of my own code
- [x] My code follows the Airbnb React Style Guidelines
- [x] Documentation has been updated to reflect my changes
- [x] New and existing tests pass locally with my changes

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
Tested locally and in containers through manual testing and with Jest tests.

### CODE REVIEW

File: api/controllers/requests-api-controller.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  purchaseDate: 1,
+  submissionDate: 1,
+  state: 1
+}
+
+// Get all request records
+export const getAllRequests = async (req: Request, res: Response) => {
+  const { minimal } = req.query;
+  try {
+    // TODO: Is there a more memory-efficient solution? Like only returning the first X records?
```
Comment (akroon3r) [inline api/controllers/requests-api-controller.ts:47]: Something like this? https://isotropic.co/how-to-implement-pagination-in-mongodb/

File: api/constants/Constants.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 const Constants = {
   HOSTNAME: process.env.HOSTNAME || 'localhost',
-  API_PORT: process.env.API_PORT || 3004
+  API_PORT: process.env.API_PORT || 3004,
+  TESTING: process.env.TESTING || false
```
Comment (akroon3r) [inline api/constants/Constants.ts:4]: A little nervous about this, but we can talk about it in person later.

**Review by akroon3r — APPROVED** [review]
Looks good at first glance, day 15 no coffee.

File: api/constants/Constants.ts (hunk 2 above)
Comment (dbarkowsky) [reply api/constants/Constants.ts:4]: 10-4

**Pull request merged**